### PR TITLE
New pattern-kinesis data stream & lambda consumer

### PR DESCRIPTION
**Pull request by awsrahulrsr**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
My email - rahulrsr@amazon.com

### CODE REVIEW

**Comment by boyney123** [comment]
Hey @awsrahulrsr ,

Missing the example-pattern json file in your pull request, once you add that we can review your pattern. Thanks

**Comment by awsrahulrsr** [comment]
Hi @boyney123, Added example-pattern.json file. Please let me know if any further changes are needed.

File: kinesis-lambda-terraform/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+    "title": "Kinesis Data Streams With Lambda Consumer",
+    "description": "When data is loaded into Kinesis data stream, the lambda consumer is invoked.",
```
Comment (boyney123) [inline kinesis-lambda-terraform/example-pattern.json:3]:
```suggestion
    "description": "When data is loaded into Kinesis data stream, the configured lambda function is invoked.",
```

File: kinesis-lambda-terraform/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "language": "",
+    "level": "200",
+    "framework": "Terraform",
+    "introBox": {
+      "headline": "How it works",
+      "text": [
+        "This pattern creates a Kinesis Data Stream and lambda consumer with event source mapping. ",
```
Comment (boyney123) [inline kinesis-lambda-terraform/example-pattern.json:10]:
```suggestion
        "This pattern creates a Kinesis Data Stream and lambda function (consumer) with event source mapping. ",
```

File: kinesis-lambda-terraform/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        "When the data is loaded into the stream, the lambda function is invoked. "
+      ]
+    },
+    "gitHub": {
+      "template": {
+        "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/kinesis-lambda-terraform",
+        "templateURL": "serverless-patterns/kinesis-lambda-terraform",
+        "projectFolder": "kinesis-lambda-terraform",
+        "templateFile": "kinesis-lambda-terraform/main.tf"
```
Comment (boyney123) [inline kinesis-lambda-terraform/example-pattern.json:19]:
```suggestion
        "templateFile": "main.tf"
```

File: kinesis-lambda-terraform/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      }
+    },
+    "resources": {
+      "bullets": [
+        {
+          "text": "Kinesis data streams with lambda consumer",
```
Comment (boyney123) [inline kinesis-lambda-terraform/example-pattern.json:25]:
```suggestion
          "text": "Kinesis Data Streams with Lambda consumers",
```

**Comment by boyney123** [comment]
Thank you @awsrahulrsr

**Pull request merged**